chapter6.adoc

```diff
@@ -232,143 +275,241 @@ specification and the upper nibble is used to hold the major version of the
 specification. For example, an implementation that supports version 1.0 of the
 specification reports 0x10.
 
-The `MPTM` field indicates the supported MPT address protection schemes. If 1,
-then the MPT modes for RV64 are supported else the MPT modes for RV32 are
-supported.
-
 The `QOSID` field indicates whether associating QoS IDs with requests is
-supported; if set to 1, the QoS ID related fields in `operand-2` are
-implemented, otherwise these fields are reserved and read-only zero.
+supported.
 
-[[CTRL]]
-=== Control register (`control`)
+[[CAP]]
+=== Control (`control`)
 
-The `control` register is used to control classification of DMA requests using
-the identifiers associated with the DMA requests to determine the associated
-IO supervisor domain ID (`IOSDID`) and the Machine-level Memory Protection Tables (`MMPT`).
+This register is used to control the operations of the I/O MPT checker.
+An implementation may allow one or more fields in the register to be
+writable to support enabling or disabling the feature controlled by that
+field. All I/O MPT checkers must support the `Off` and `On` values for
+the `MODE` field.
 
-[caption="Register {counter:rimage}: ", reftext="Register {rimage}", title="Control register (`control`)"]
+.Control Register (`control`)
 [wavedrom, , ]
 ....
 {reg: [
-  {bits:  8, name: 'OP (WARL)'},
-  {bits:  8, name: 'RULEID (WARL)'},
-  {bits: 16, name: 'WPRI'},
-  {bits:  7, name: 'STATUS (RO)'},
+  {bits:  4, name: 'MODE (WARL)'},
+  {bits:  1, name: 'MBE (WARL)'},
+  {bits:  1, name: 'MXL (WARL)'},
+  {bits: 26, name: 'WPRI'},
   {bits:  1, name: 'BUSY (RO)'},
-  {bits: 24, name: 'WPRI'}
-], config:{lanes: 8, hspace:1024}}
+], config:{lanes: 2, hspace:1024}}
 ....
 
-The `OP` field is used to instruct IOMPTCHK to perform an operation listed in
-<<IOMPTCHK_OP>>. The `RULEID` is identifier of a rule in the SDCL function to
-operate on. The `RULEID` value of 0 indicates that the operation applies to all
-rules and is supported only if explicitly specified by an operation.
+The I/O MPT checker may be configured to be in the following modes:
+
+[[IOMPTCHK_MODE]]
+.I/O MPT checker modes (`MODE`)
+[width=100%]
+[%header, cols="16,^12,70"]
+|===
+| Name         | Encoding ^| Description
+| Off          | 0         | No inbound memory transactions are allowed.
+| On           | 1         | I/O MPT checker is operational.
+| Reserved     | 2-13      | Reserved for future standard use.
+| Custom       | 14-15     | Designated for custom use.
+|===
+
+When the `MODE` field value is changed to `Off` the I/O MPT checker guarantees
+that in-flight transactions, observed at the time of the write to this field,
+from devices connected to the I/O MPT checker will either be processed with the
+configurations applicable to the old value of the `MODE` field or are aborted
```

Comment:
why provide this option - since there is an IOFENCE command - perhaps this should simply say abort any in-flight?

Reply:
An implementation may be "past point of no return on handling" in which case it is allowed to complete it. Else it can abort it.